FLT/Deformations/Subfunctor.lean

```diff
@@ -0,0 +1,213 @@
+-- **Modified from `Subpresheaf.lean` in mathlib**
```

Comment:
Looks like the file [CategoryTheory.Subpresheaf](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/Subpresheaf/Basic.html#CategoryTheory.Subpresheaf). Probably we should just replace Subpresheaf by Subfunctor? So this file should go into the Mathlib folder?

Reply:
Indeed. I have not yet made up my mind if this should replece it or if we should have both.
And hence I also prefer if this file is not touched to keep its similarity with the original file despite that your reviews to this file makes a lot of sense.